### PR TITLE
Logout changes

### PR DESCRIPTION
**Pull request by willsb**
## What's this?
This unschedules notifications when logging out

## Why do we want this?
Because reminders should be handled only when the app is in the logged in state

## How is it done?
We simply run the service directly. Ideally, this should all be an interactor. I'll push for a bigger refactor later.

## :squid: Permissions
Anyone, anytime

### CODE REVIEW

File: Toggl.Foundation.Tests/DataSources/TogglDataSourceTests.cs [hunk 1]
```diff
@@ -89,6 +91,19 @@ public void DoesNotClearTheDatabaseBeforeTheSyncManagerCompletesFreezing()
                 Database.DidNotReceive().Clear();
             }
 
+            [Fact, LogIfTooSlow]
+            public void UnscheduleAllNotifications()
```
Comment (sofokli1) [inline Toggl.Foundation.Tests/DataSources/TogglDataSourceTests.cs]: Based on the above tests, shouldn't this be called `UnschedulesAll...`?

File: Toggl.Foundation.Tests/DataSources/TogglDataSourceTests.cs [hunk 2]
```diff
@@ -89,6 +91,19 @@ public void DoesNotClearTheDatabaseBeforeTheSyncManagerCompletesFreezing()
...
+            {
+                var scheduler = new TestScheduler();
+                SyncManager.Freeze().Returns(Observable.Never<SyncState>());
+
+                var observable = DataSource.Logout().SubscribeOn(scheduler).Publish();
+                observable.Connect();
+                scheduler.AdvanceBy(TimeSpan.FromDays(1).Ticks);
+
+                NotificationService.DidNotReceive().UnscheduleAllNotifications();
```
Comment (sofokli1) [inline Toggl.Foundation.Tests/DataSources/TogglDataSourceTests.cs]: I don't get this. why `DidNotReceive`? shouldn't it receive `UnscheduleAllNotifications` based on the name of the test?

File: Toggl.Foundation/DataSources/TogglDataSource.cs [hunk 3]
```diff
@@ -146,6 +144,10 @@ public IObservable<Unit> Logout()
                 .Do(_ => stopSyncingOnSignal())
                 .SelectMany(_ => database.Clear())
                 .Do(_ => shortcutCreator.OnLogout())
+                .SelectMany(_ => 
+                    notificationService
+                        .UnscheduleAllNotifications()
+                        .Catch(Observable.Return(Unit.Default)))
```
Comment (juanlaube) [inline Toggl.Foundation/DataSources/TogglDataSource.cs]: In what particular situation can this throw? 🤔

Comment (willsb) [reply Toggl.Foundation/DataSources/TogglDataSource.cs]: If we don't have the permission for notifications

File: Toggl.Foundation.Tests/DataSources/TogglDataSourceTests.cs (hunk 2 above)
Comment (willsb) [reply Toggl.Foundation.Tests/DataSources/TogglDataSourceTests.cs]: ©️ 🍝

**Pull request merged**